components/create_post/create_post.jsx

```diff
@@ -887,9 +914,40 @@ export default class CreatePost extends React.Component {
 
         let serverError = null;
         if (this.state.serverError) {
+            let serverErrorContent = this.state.serverError.message;
+
+            if (isErrorInvalidSlashCommand(this.state.serverError)) {
+                const storedMessage = this.state.serverError.submittedMessage;
+                const command = storedMessage.split(' ')[0];
+
+                serverErrorContent = (
+                    <FormattedMessage
+                        id='create_post.invalidCommand'
+                        defaultMessage={'Command with a trigger of \'{command}\' not found. {send_as_message}'}
+                        values={{
+                            command,
+                            send_as_message: (
+                                <a
+                                    ref='sendAsMessageLink'
+                                    href='#'
+                                    onClick={this.handleSubmit}
+                                    className='create-post__send-as-message'
+                                >
+                                    <FormattedMessage
+                                        id='create_post.sendAsMessageLink'
+                                        defaultMessage='Click here to send as a message.'
+                                    />
+                                </a>
+                            ),
+                        }}
+                    />
```

Comment:
Can we move this into a separate stateless component and be tested?

Reply:
For sure, are you thinking the whole `if (this.state.serverError) {` block should be extracted? I think that makes the most sense because otherwise the extracted component wouldn't be doing any logic. This would open up more possibilities for modularity around post submit errors. And, as you commented:

> I think the same should be implemented in the CreateComment or RHS thread

The generic error message component could be reused in that context as well. It can know how to display itself, given the error that has been passed in.

Reply:
Right, I think that makes sense.

Reply:
If the component were to be used with `posts` and `comments`, is there a common name to be used among them? Something like `SubmitError`

Reply:
I'm not good at naming but I guess it sounds good :)

Reply:
As I think more about it, `SubmitError` is too vague and could be referring to any form in the application where you are submitting something. I'm thinking `MessageSubmitError` would be a better name. Either way, the component needs to be implemented, so I'll get on that :)

Reply:
👍

Reply:
I've made the change to extract the error message component. If you would like, I can make a PR for the error to be used in the `create_comment` component once this is merged.

Reply:
Sure, sounds good.